src/typechat.ts

```diff
@@ -42,6 +42,15 @@ export interface TypeChatJsonTranslator<T extends object> {
      * @returns A repair prompt constructed from the error message.
      */
     createRepairPrompt(validationError: string): string;
+    /**
+     * Optionally implements additional validation logic beyond what is expressed in the schema. This function is
+     * called following successful schema validation of an instance. By default the function just returns a
+     * `Success<T>` for the given instance, but an application can assign a new function that implements any
+     * additional validation.
+     * @param instance The instance to validate.
+     * @returns A `Success<T>` with the final validated instance, or an `Error` explaning the validation failure.
+     */
+    validateInstance(instance: T): Result<T>;
```

Comment:
Aside - because you mention that this is "optional", it's a little weird that it's always present.

Also, it is a little odd that this API really expects people to assign directly to the result object. You can't do something like

```ts
const translator = {
  ...createJsonTranslator(/*...*/),
  validateInstance: /*...*/,
}
```

Reply:
Well, it's optional to implement it, but that doesn't mean there isn't a default implementation. Right, you can't just spread like that, the methods are bound to the particular instance (i.e. they don't use `this`).